rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -217,12 +220,9 @@ std::vector<rcl_interfaces::msg::SetParametersResult>
 Node::set_parameters(
   const std::vector<rclcpp::parameter::ParameterVariant> & parameters)
 {
-  std::lock_guard<std::mutex> lock(mutex_);
   std::vector<rcl_interfaces::msg::SetParametersResult> results;
   for (auto p : parameters) {
-    parameters_[p.get_name()] = p;
-    rcl_interfaces::msg::SetParametersResult result;
-    result.successful = true;
+    auto result = set_parameters_atomically({{p}});
     // TODO: handle parameter constraints
```

Comment:
This todo can be removed. it's now redundant with https://github.com/ros2/rclcpp/pull/39/files#diff-809285dba1df7726afedd63625b6a6ceR260 

Code reuse saving us todos!